pkg/controller/infinispan/resources/config/xsite.go

```diff
@@ -20,51 +20,51 @@ import (
 )
 
 // ComputeXSite compute the xsite struct for cross site function
-func ComputeXSite(infinispan *ispnv1.Infinispan, kubernetes *kube.Kubernetes, service *corev1.Service, logger logr.Logger, xsite *config.XSite) error {
-	if infinispan.HasSites() {
-		siteServiceName := infinispan.GetSiteServiceName()
-		localSiteHost, localSitePort, err := getCrossSiteServiceHostPort(service, kubernetes, logger)
-		if err != nil {
-			logger.Error(err, "error retrieving local x-site service information")
-			return err
-		}
+func ComputeXSite(infinispan *ispnv1.Infinispan, kubernetes *kube.Kubernetes, service *corev1.Service, logger logr.Logger) (error, *config.XSite) {
+	siteServiceName := infinispan.GetSiteServiceName()
+	localSiteHost, localSitePort, err := getCrossSiteServiceHostPort(service, kubernetes, logger)
+	if err != nil {
+		logger.Error(err, "error retrieving local x-site service information")
+		return err, nil
```

Comment:
Is it necessary to log this here and then again in `Process()`?

I think it would be better to return an update error here, then only log once in `Process()`. E.g.:

```
	if err != nil {
		return fmt.Errorf("Error retrieving local x-site service information: %w", err), nil
	}
```

Reply:
Agreed, @ryanemerson 
Saw, that xsite and other components logging need to be improved, let's do this in separate issue you opened